src/forecasts.jl

```diff
@@ -1,443 +1,144 @@
-abstract type Forecast <: InfrastructureSystemsType end
 
-const ForecastComponentLabelPair = Tuple{<:InfrastructureSystemsType, String}
-const ForecastComponentLabelPairByInitialTime = Dict{Dates.DateTime,
-                                                     Set{ForecastComponentLabelPair}}
-const UNITIALIZED_DATETIME = Dates.DateTime(Dates.Minute(0))
-const UNITIALIZED_PERIOD = Dates.Period(Dates.Minute(0))
-const UNITIALIZED_HORIZON = 0
+"""
+Abstract type for forecasts that are stored in a system.
+Users never create them or get access to them.
+Stores references to time series data, so a disk read may be required for access.
+"""
+abstract type ForecastInternal <: InfrastructureSystemsType end
+
 
+"""
+Abstract type for forecasts supplied to users. They are not stored in a system. Instead,
+they are generated on demand for the user.
+Users can create them. The system will convert them to a subtype of ForecastInternal for 
+storage.
+Time series data is stored as a field, so reads will always be from memory.
+"""
+abstract type Forecast <: Any end
 
+get_horizon(forecast::Forecast) = length(get_data(forecast))
```

Comment:
if get `get_data` returns the whole time_series then the results won't be consistent with the internal horizon value?

Reply:
I think this was resolved in our conversation.